Added lerp_angle() which takes the shortest path between two angles

tristanbob opened the pull request:
- This is my first commit to the core, so I'm unsure of its quality
- I copied another commit to understand what needed changed
- I have not tested this function (let me know of a good way to do this)

AlexandreSi commented on Core/GDCore/Extensions/Builtin/MathematicalToolsExtension.cpp:
```diff
@@ -492,6 +492,16 @@ BuiltinExtensionsImplementer::ImplementsMathematicalToolsExtension(
       .SetHelpPath("/all-features/expressions");
+
+  extension
+      .AddExpression("lerp_angle",
```
Can you use camelCase for the expression and function names? There's currently a lack of consistency across all expressions, but I think that camelCase is the most used.

AlexandreSi commented on GDJS/GDJS/Extensions/Builtin/MathematicalToolsExtension.cpp:
```diff
@@ -71,6 +71,7 @@ MathematicalToolsExtension::MathematicalToolsExtension() {
   GetAllExpressions()["XFromAngleAndDistance"].SetFunctionName("gdjs.evtTools.common.getXFromAngleAndDistance");
   GetAllExpressions()["YFromAngleAndDistance"].SetFunctionName("gdjs.evtTools.common.getYFromAngleAndDistance");
   GetAllExpressions()["Pi"].SetFunctionName("gdjs.evtTools.common.pi");
+  GetAllExpressions()["lerp_angle"].SetFunctionName("gdjs.evtTools.common.lerp_angle");
```
camelCase to use here as well

AlexandreSi commented on GDJS/Runtime/events-tools/commontools.ts:
```diff
@@ -371,3 +371,18 @@ namespace gdjs {
+        angle1: number,
+        angle2: number,
+        x: float
+      ): number {
```
In typescript (javascript), `number`, `float` and `integer` are the same.
It can help a human understand some things though. Here I would use `float` as return type since `x` is supposed to be a float.

AlexandreSi commented on GDJS/Runtime/events-tools/commontools.ts:
```diff
@@ -371,3 +371,18 @@ namespace gdjs {
+       * @param x The interpolation value between 0 and 1.
+       * @returns Return the interpolated angle, in degrees.
+       */
+       export const lerp_angle = function (
```
This function should be in the namespace `gdjs.evtTools.common`, just below `pi`

AlexandreSi reviewed (commented):
Thanks for this contribution! I could test it on my laptop and works fine when the new function is in the correct namespace

AlexandreSi commented:
You might need to run auto formatting in the GDJS file as well. DM me if you need any help for that.

tristanbob commented:
@AlexandreSi I think I addressed all your suggestions. Thanks for coaching me on this!  

Please let me know if there are any other things that need to be changed on this PR.  :)

tristanbob commented:
@AlexandreSi how hard would it be to make a build with this change? I have not been able to test the new function yet.  

If you can provide a build I will make a simple example game to test it.

AlexandreSi commented:
LGTM But there's a formatting issue. I'll use your code to push a version that passes the tests so that it's in the upcoming version

AlexandreSi commented:
> @AlexandreSi how hard would it be to make a build with this change? I have not been able to test the new function yet.
> 
> If you can provide a build I will make a simple example game to test it.

It would take some time. I already tested it and it seemed fine to me. You'll be able to test it when the new version is available and if anything weird happens, we'll fix or revert it.
